cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractDistributedProgramRunner.java

```diff
@@ -151,6 +158,7 @@ protected AbstractDistributedProgramRunner(TwillRunner twillRunner, YarnConfigur
     this.cConf = cConf;
     this.eventHandler = createEventHandler(cConf);
     this.secureStoreUpdater = tokenSecureStoreUpdater;
+    this.fileContextLocationFactory = new FileContextLocationFactory(hConf);
```

Comment:
You don't need a new one. Just use the regular CDAP `LocationFactory` coming through `@Inject`. If you need everything relative to the root, then inject the `RootLocationFactory`.